### ISSUE TITLE
The Item drop from boss in arena will remain after the arena end

### ISSUE THREAD

**Issue by congtubot** [garbagemule/MobArena#556]
# Feature request

**Short description**

Yah . Like if a monster drop the a weapon . Then when you leave arena , the weapon still remain in your inventory . And .. the item you use in arena will lost . It like the inventory in arena and before/after join inventory is one

**Implementation details**

It should have a config line like Share-item-world . It will remove the class myitems , because all inventory will be one . 

This will make the game more fair , because in some case , a player dont do anything still have reward . And with mythicmobs , it is easy to drop some custom item , Very needed to a RPG server

**Comment by garbagemule** [comment]
If I understand it correctly, this feature request is about treating drops as "rewards" in the sense that they are kept on arena end. Because of the way the code base is structured right now, this is not feasible to try to implement before the sessions rework is over. However, there are some things we need to address before we can start work on this.

Part of the motivation is that "lazy" players are given rewards that they didn't really earn. Some points to consider:

- If 10 people are playing and only 1 player is lazy, having drops as rewards means that only 1 out of 9 non-lazy players is rewarded. The remaining 8 non-lazy players are punished as if they weren't doing anything. This is arguably more unfair.
- If you have to be in range to catch rewards, it means that ranged classes will almost never be able to catch the drops before the melee classes pick them up. This means we're removing incentive to play ranged classes, which is hardly desirable in the general case.

There is another issue with this request, and it's in the assumption that "My Items" is just "one" inventory that transcends the arena sessions. This is _not_ true. All inventory items are restored on arena end, even if using the "My Items" class. But this opens up a new potential feature: being able to use "My Items" in a way that the inventory is only saved on death and restored on respawn. This means that if I join with 10 health potions and use 3, then die, I'll come out of the arena with 7 health potions, even though I died. In a "hardmode" version, I may leave with an empty inventory.
